cdap-app-fabric/src/main/java/co/cask/cdap/test/RuntimeStats.java

```diff
@@ -154,37 +157,32 @@ private void doWaitFor(String name, long count, long timeout, TimeUnit timeoutUn
       @Override
       public String toString() {
         return String.format("%s; input=%d, processed=%d, exception=%d",
-                             prefix, getInput(), getProcessed(), getException());
+                             Joiner.on(",").withKeyValueSeparator(":").join(context),
+                             getInput(), getProcessed(), getException());
       }
     };
   }
 
-  public static void clearStats(final String prefix) {
-    Iterators.removeIf(counters.entrySet().iterator(), new Predicate<Map.Entry<String, AtomicLong>>() {
-      @Override
-      public boolean apply(Map.Entry<String, AtomicLong> input) {
-        return input.getKey().startsWith(prefix);
+  private static long getTotalCounter(Map<String, String> context, String metricName) {
+    CubeQuery query = getTotalCounterQuery(context, metricName);
+    try {
+      Collection<TimeSeries> result = metricStore.query(query);
+      if (result.isEmpty()) {
+        return 0;
       }
-    });
+      // since it is totals query and not groupBy specified, we know there's one time series with one time value
+      return result.iterator().next().getTimeValues().get(0).getValue();
```

Comment:
do we need a hasNext check before this accessing the time-series ? if the data is not yet collected we may not get a time series value yet and we we want to retry before throwing exception ?